source/menu/menu_coop.qc

```diff
@@ -0,0 +1,299 @@
+
+	Menu_DrawDivider(3);
+	Menu_Button(3.25, "cdm_connect", "CONNECT TO SERVER", "Attempt to connect to Game.") ? localcmd("connect ", server_ip, "\n") : 0;
+
+	Menu_Button(-1, "cdm_back", "BACK", "Return to Join Game Menu.") ? current_menu = MENU_COOPJOIN : 0;
+
+	sui_pop_frame();
+};
+
+void() Menu_Coop_Join =
+{
+	Menu_DrawBackground();
+
+	Menu_DrawTitle("JOIN GAME");
+
+	Menu_Button(1, "cjm_browse", "BROWSE SERVERS", "Browse the list of active Servers.") ? current_menu = MENU_COOPBROWSE : 0;
+	Menu_Button(2, "cjm_direct", "DIRECT IP/ID", "Connect directly to an active Server.") ? current_menu = MENU_COOPDIRECT : 0;
+
+	Menu_Button(-1, "cjm_back", "BACK", "Return to Co-Op Menu.") ? current_menu = MENU_COOP : 0;
+
+	sui_pop_frame();
+};
+
+void() Menu_Coop_Create =
+{
+	Menu_Coop_Hostname_Update();
+	Menu_Coop_Port_Update();
+	Menu_Coop_Password_Update();
+
+	Menu_DrawBackground();
+
+	Menu_DrawTitle("CREATE GAME");
+
+	Menu_Button(1, "ccm_servername", "SERVER NAME", "Enter a name that will appear in the Server List.");
+	Menu_TextInput("ccm_servername", 1, server_hostname, server_hostname_cursor);
+
+	Menu_Button(2, "ccm_port", "SERVER PORT", "Edit the port that the Server will use.");
+	Menu_TextInput("ccm_port", 2, server_port, server_port_cursor);
+
+	Menu_Button(3, "ccm_password", "SERVER PASSWORD", "Enter an (optional) password to limit who can join.");
+	Menu_PasswordInput("ccm_password", 3, server_password, server_password_cursor);
+
+	Menu_DrawDivider(4);
+	Menu_Button(4.25, "ccm_choosemap", "CHOOSE MAP", "Select a Map to start the Game.") ? Menu_StartCoop() : 0;
+
+	Menu_Button(-1, "ccm_back", "BACK", "Return to Co-Op Menu.") ? current_menu = MENU_COOP : 0;
+
+	sui_pop_frame();
+};
+
+void() Menu_Coop =
+{
+	Menu_Coop_Name_Update();
+
+	Menu_DrawBackground();
+
+	Menu_DrawTitle("COOPERATIVE");
+
+	Menu_Button(1, "cm_playername", "PLAYER NAME", "Name that appears in-game and on the Scoreboard.");
+	Menu_TextInput("cm_playername", 1, player_name, player_name_cursor);
+
+	Menu_DrawDivider(2);
+	Menu_Button(2.25, "cm_join", "JOIN GAME", "Join an in-progress Match.") ? current_menu = MENU_COOPJOIN : 0;
+	Menu_Button(3.25, "cm_create", "CREATE GAME", "Create a new Match for others to join.") ? current_menu = MENU_COOPCREATE : 0;
+
+	Menu_Button(-1, "cm_back", "BACK", "Return to Main Menu.") ? current_menu = MENU_MAIN : 0;
+
+	sui_pop_frame();
+};
+
+string menu_coop_browse_buttons[] = {
+	"cbm_serverlist", "cbm_refresh", "cbm_back"
+};
+
+string menu_coop_direct_buttons[] = {
+	"cdm_serverip", "cdm_serverpassword", "cdm_connect", "cdm_back"
+};
+
+string menu_coop_join_buttons[] = {
+	"cjm_browse", "cjm_direct", "cjm_back"
+};
+
+string menu_coop_create_buttons[] = {
+	"ccm_servername", "ccm_port", "ccm_password", "ccm_choosemap", "ccm_back"
+};
+
+string menu_coop_buttons[] = {
+	"cm_playername", "cm_join", "cm_create", "cm_back"
+};
+
+// can you tell i'm a C user at heart? -- erysdren
```

Comment:
Not sure how I feel about these blocks, if only because if later down the line a dynamic button needs added it severely complicates things. In places like the main menu you can see the logic in place to hide things like the quit button while still being able to scroll fine with kb/gamepad